### PR TITLE
feat: add marked and flagged as duplicate to all applications table

### PR DESCRIPTION
**Pull request by KrissDrawing**
# Pull Request Template

## Issue Overview

This PR addresses #3103

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Added two columns `flagged as duplicate`, and `marked as duplicate` to All Aplications.
Marked as duplicate has its own field. Flagged as duplicate is when reviewStatus equals 'duplicate'

## How Can This Be Tested/Reviewed?

Go to applications tab in listing. In All Application table scroll to the right. Last two columns should be `Flagged as Duplicate`, and `Marked as Duplicate`. On null / undefined state they should display n/a (not sure if possible)

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/applications/ApplicationsColDefs.ts
Comment (ColinBuyck) [inline sites/partners/src/applications/ApplicationsColDefs.ts:593]: Only thing I noticed on this PR was that I believe this should return yes for either "pending" or "duplicate" rather than just "duplicate". The way I'm understanding the difference between these columns is that an application may be flagged as a duplicate automatically but not yet marked as a duplicate by the user which results in the pending state. Is this correct @kathyccheng?

Comment (YazeedLoonat) [reply sites/partners/src/applications/ApplicationsColDefs.ts:593]: `flagged as duplicate` means that the application is marked as a possible duplicate
`marked as duplicate`  (when the review status === duplicate) is when the application was confirmed as a duplicate

Comment (emilyjablonski) [reply sites/partners/src/applications/ApplicationsColDefs.ts:593]: Has this been resolved?

Comment (ColinBuyck) [reply sites/partners/src/applications/ApplicationsColDefs.ts:593]: I don't believe so @emilyjablonski.  If I am understanding our conversations correctly @YazeedLoonat, the issue is that flagged as duplicated will become false if a flagged application is then marked as valid when using the review_status. I believe there are three cases where flagged as duplicate should be true: 1) Application is part of a duplicate group in pending review (would be covered by my comment above) 2) Application was part of a duplicate group in pending review but then marked as duplicate covered by PR currently) 3) Application was part of a duplicate group in pending review but then marked as valid which cannot currently be represented by the review_status. Let me know if this makes sense @KrissDrawing and what your thoughts on next steps would be @YazeedLoonat.

**Comment by KrissDrawing** [comment]
@YazeedLoonat Thanks! 
Flagged looks good, but `markedAsDuplicate` field still doesn't seem to return correct values.
Even after resolving application (so its reviewStatus = 'duplicate') field `markedAsDuplicate` = false.
I can see in exported csv file it is properly marked as duplicate 🤔 

On frontend i can solve that just by checking reviewStatus. But then don't know what to do with that `markedAsDuplicate` field as it might be misleading in the future i guess 😅

**Comment by YazeedLoonat** [comment]
Hey @KrissDrawing got this updated for ya, should be good to go now hopefully if you wanna take another look

**Comment by KrissDrawing** [comment]
@YazeedLoonat Works Good 🎉 
@ColinBuyck @emilyjablonski it's ready to review

File: backend/core/src/applications/services/applications.service.ts
Comment (ludtkemorgan) [inline backend/core/src/applications/services/applications.service.ts:118]: I wonder if this should be done after the check for user authorization so we avoid doing the duplicate db call if the user isn't even authorized. @YazeedLoonat thoughts?

**Review by ludtkemorgan — CHANGES_REQUESTED** [review]
Functionally it is working as expected. Looks like there are failing tests now though. https://app.circleci.com/pipelines/github/bloom-housing/bloom/9194/workflows/37fff7e6-8ad8-42ad-bb22-51d226b8f5b7/jobs/66264. Can you look into those?

File: backend/core/src/applications/services/applications.service.ts
Comment (YazeedLoonat) [reply backend/core/src/applications/services/applications.service.ts:118]: just to add some notes here from yesterday's discussion:
I think we do want to do that but for right now its probably okay to leave this as is. 

We more or less need to undertake an auth investigation and this can fall into that investigation bucket

**Review by ludtkemorgan — APPROVED** [review]

**Comment by emilyjablonski** [comment]
@YazeedLoonat Is this one ready?

**Comment by YazeedLoonat** [comment]
Hey @emilyjablonski I think its ready for another set of eyes I probably shouldn't be a reviewer on this since I committed code to this pr though

**Pull request merged**
